app/models/product.rb

```diff
@@ -15,6 +15,9 @@ class Product < ActiveRecord::Base
   has_many   :price_policies
   has_many   :training_requests, dependent: :destroy
 
+  cattr_accessor(:types) { ConfigurableArray.new([Instrument, Item, Service, Bundle]) }
```

Comment:
I don't entirely follow why we need a special `ConfigurableArray` class rather than using a vanilla array. Is the order so important that we can't just use `.unshift` or `<<`?